tests/integration/wordpress_client.py

```diff
@@ -0,0 +1,122 @@
+import re
+import json
+
+import requests
+
+
+class WordpressClient:
+    """A very simple WordPress client for test purpose only"""
+
+    def __init__(self, host: str, username: str, password: str, is_admin: bool):
+        self.host = host
+        self.username = username
+        self.password = password
+        self._session = requests.session()
+        self.timeout = 10
+        self._login()
+        try:
+            self._session.get(
+                f"http://{self.host}/wp-json/",
+                timeout=self.timeout
+            ).json()
+        except requests.exceptions.JSONDecodeError:
```

Comment:
For my understanding, could you explain what is going on here please?

Reply:
WordPress will not expose the wp-json endpoints by default, here, the client check if wp-json is exposed, if not change the permalink setting to expose that.

Reply:
Ah that makes sense, do you think it is worth adding a comment to explain that? I was quite confused before and that explanation helped a lot 😄

Reply:
Added, thanks.